### PR TITLE
[s2s] test_bash_script.py - actually learn something

### LINKED ISSUE #6049: examples/seq2seq/test_bash_script.py :: actually learn something
At the moment validation bleu barely gets above zero in the tests, so they don't really prove much about our code.

we could use a larger model like sshleifer/student_marian_6_3, and more data, and train for 10 minutes . This would allows us to test whether changing default parameters/batch techniques obviously degrades performance.

The github actions CI reuses it's own disk, so this will only run there and hopefully not have super slow downloads.

### PR DESCRIPTION
**Pull request by stas00**
As discussed in https://github.com/huggingface/transformers/issues/6049 this PR replaces the original `test_train_mbart_cc25_enro_script` test with mostly new guts doing the following:

* [x] switches to downloading and caching a bigger custom dataset, which is a subset of the full wmt_en_ro.  I created https://cdn-datasets.huggingface.co/translation/wmt_en_ro-tr40k-va0.5k-te0.5k.tar.gz - hope the name is intuitive - self-documenting. It's just 3.6M (vs 56M original). I made it using this script: https://github.com/stas00/porting/blob/master/transformers/translation/make-wmt_en_ro-subset.md

* [x] performs qualitative checks on the eval/test results:
   - Minimum learning requirement 1: BLEU improves over the course of training by more than 2 pts
   - Minimum learning requirement 2: BLEU finishes above 17
   - Minimum learning requirement 3: test bleu and val bleu within ~1 pt. 

also makes some extra improvements:
* [x] uses `require_torch_gpu` decorator
* [x] removes hardcoded paths

Runtime is about ~2min on unoptimized rtx-3090.

Fixes: #6049

@sshleifer

### CODE REVIEW

File: examples/seq2seq/test_bash_script.py [hunk 1]
```diff
@@ -3,92 +3,118 @@
...
     @slow
-    @pytest.mark.skipif(not CUDA_AVAILABLE, reason="too slow to run on CPU")
+    @require_torch_gpu
     def test_train_mbart_cc25_enro_script(self):
-        data_dir = "examples/seq2seq/test_data/wmt_en_ro"
+        DEBUG = 1  # XXX
+        data_dir = self.data_dir
         env_vars_to_replace = {
             "--fp16_opt_level=O1": "",
             "$MAX_LEN": 128,
```
Comment (sshleifer) [inline examples/seq2seq/test_bash_script.py:48]: MAX_LEN=64 might speed things up?

**Review by sshleifer — APPROVED** [review]
LGTM, thanks. LMK when ready to merge!

File: examples/seq2seq/test_bash_script.py (hunk 1 above)
Comment (stas00) [reply examples/seq2seq/test_bash_script.py:48]: 128: 3min, 64: 2:50m - the bleu scores for val/test are within 0.1 difference

Switch anyway as every second adds up?

Comment (sshleifer) [reply examples/seq2seq/test_bash_script.py:48]: Yep!

**Comment by stas00** [comment]
It's ready to merge

**Pull request merged**